notification-config-service-impl/src/main/java/org/hypertrace/notification/config/service/NotificationConfigServiceImpl.java

```diff
@@ -0,0 +1,230 @@
+import org.hypertrace.notification.config.service.v1.CreateNotificationRuleResponse;
+import org.hypertrace.notification.config.service.v1.DeleteNotificationChannelRequest;
+import org.hypertrace.notification.config.service.v1.DeleteNotificationChannelResponse;
+import org.hypertrace.notification.config.service.v1.DeleteNotificationRuleRequest;
+import org.hypertrace.notification.config.service.v1.DeleteNotificationRuleResponse;
+import org.hypertrace.notification.config.service.v1.GetAllNotificationChannelsRequest;
+import org.hypertrace.notification.config.service.v1.GetAllNotificationChannelsResponse;
+import org.hypertrace.notification.config.service.v1.GetAllNotificationRulesRequest;
+import org.hypertrace.notification.config.service.v1.GetAllNotificationRulesResponse;
+import org.hypertrace.notification.config.service.v1.NewNotificationRule;
+import org.hypertrace.notification.config.service.v1.NotificationChannel;
+import org.hypertrace.notification.config.service.v1.NotificationChannelMutableData;
+import org.hypertrace.notification.config.service.v1.NotificationConfigServiceGrpc;
+import org.hypertrace.notification.config.service.v1.NotificationRule;
+import org.hypertrace.notification.config.service.v1.NotificationRuleMutableData;
+import org.hypertrace.notification.config.service.v1.UpdateNotificationChannelRequest;
+import org.hypertrace.notification.config.service.v1.UpdateNotificationChannelResponse;
+import org.hypertrace.notification.config.service.v1.UpdateNotificationRuleRequest;
+import org.hypertrace.notification.config.service.v1.UpdateNotificationRuleResponse;
+
+@Slf4j
+public class NotificationConfigServiceImpl
+    extends NotificationConfigServiceGrpc.NotificationConfigServiceImplBase {
+  private final NotificationChannelStore notificationChannelStore;
+  private final NotificationRuleStore notificationRuleStore;
+  private final NotificationConfigServiceRequestValidator notificationConfigServiceRequestValidator;
+
+  public NotificationConfigServiceImpl(Channel channel) {
+    this.notificationChannelStore = new NotificationChannelStore(channel);
+    this.notificationRuleStore = new NotificationRuleStore(channel);
+    this.notificationConfigServiceRequestValidator =
+        new NotificationConfigServiceRequestValidator();
+  }
+
+  @Override
+  public void createNotificationRule(
+      CreateNotificationRuleRequest request,
+      StreamObserver<CreateNotificationRuleResponse> responseObserver) {
+    try {
+      RequestContext requestContext = RequestContext.CURRENT.get();
+      notificationConfigServiceRequestValidator.validateCreateNotificationRuleRequest(
+          requestContext, request);
+      NewNotificationRule newNotificationRule = request.getNewNotificationRule();
+      NotificationRule.Builder builder =
+          NotificationRule.newBuilder()
+              .setId(UUID.randomUUID().toString())
+              .setNotificationRuleData(
+                  NotificationRuleMutableData.newBuilder()
```

Comment:
This no longer needs the mapping with the mutable data approach right? Can we set `setNotificationRuleData(newNotificationRule.getNotificationRuleData())`